src/transaction.cpp

```diff
@@ -1960,7 +1960,7 @@ int wsrep::transaction::append_sr_keys_for_commit()
                  j(i->second.begin());
              ret == 0 && j != i->second.end(); ++j)
         {
-            wsrep::key key(wsrep::key::shared);
+            wsrep::key key(wsrep::key::reference);
```

Comment:
Reference key is probably too strong here. The purpose of adding shared keys to commit fragment is to create dependency, but changing it to reference key will create a conflict with exclusive.